lib/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -189,57 +239,71 @@ namespace Opm {
         }
     }
 
-    void TimeMap::addFromDATESKeyword( const DeckKeyword& DATESKeyword ) {
+    std::time_t TimeMap::timeTFromEclipse(const DeckRecord &dateRecord) {
+        static const std::string errorMsg("The datarecord must consist of the four values "
```

Comment:
Implement a `TimeMap::mkdate_time()` based on `util_set_datetime_values_utc()` similar to the `TimeMap::mkdate()`